rocAL/source/pipeline/tensor.cpp

```diff
@@ -202,8 +209,49 @@ void Tensor::update_tensor_roi(const std::vector<uint32_t> &width,
             }
         }
     }
+    else if(!_info.is_metadata()) { // Audio - Data
+        auto max_dims = _info.max_shape();
+        unsigned max_samples = max_dims.at(0);
+        unsigned max_channels = max_dims.at(1);
+        auto samples = width;
+        auto channels = height;
+        Roi2DCords *roi = _info.roi().get_2D_roi();
+
+        if (samples.size() != channels.size())
+            THROW("Batch size of Tensor height and width info does not match")
+        if (samples.size() != info().batch_size())
+            THROW("The batch size of actual Tensor height and width different from Tensor batch size " + TOSTR(samples.size()) + " != " + TOSTR(info().batch_size()))
+        for (unsigned i = 0; i < info().batch_size(); i++) {
+            if (samples[i] > max_samples) {
+                ERR("Given ROI width is larger than buffer width for tensor[" + TOSTR(i) + "] " + TOSTR(samples[i]) + " > " + TOSTR(max_samples))
+                roi[i].xywh.x = max_samples;
+            }
+            else {
+                roi[i].xywh.x = samples[i];
```

Comment:
Also we discussed we can have it in w and h instead of x and y positions please check and make the change

If this is changed should be changed for other augmentations as well please note @swetha097 and @SundarRajan28